daemon/cmd/daemon.go

```diff
@@ -1065,13 +1059,29 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 	// router IPs from the K8s resources if we weren't able to restore them
 	// from the fs. We must do this after IPAM because we must wait until the
 	// K8s resources have been synced. Part 2/2 of restoration.
-	if option.Config.EnableIPv4 {
-		d.restoreCiliumHostIPs(false, router4FromK8s)
-	}
-	if option.Config.EnableIPv6 {
-		d.restoreCiliumHostIPs(true, router6FromK8s)
+	bo := wait.Backoff{
+		Duration: 100 * time.Millisecond,
+		Factor:   1,
+		Jitter:   0.1,
+		Steps:    3,
+	}
+	err = wait.ExponentialBackoffWithContext(ctx, bo, func(ctx context.Context) (done bool, err error) {
```

Comment:
I feel like this boilerplate should ideally be collapsed down into a helper function, so that the daemon code doesn't just increasingly become longer each time we add resiliency to the restore paths.

Reply:
Indeed! I'll float an idea here and we can backfill the other prs if the proposal is deemed acceptable...